process_report/invoices/pi_specific_invoice.py

```diff
@@ -43,31 +70,109 @@ def _prepare(self):
         ]
         self.pi_list = self.export_data[invoice.PI_FIELD].unique()
 
+    def _get_pi_dataframe(self, data, pi):
+        def add_dollar_sign(data):
+            if pandas.isna(data):
+                return data
+            else:
+                return "$" + str(data)
+
+        pi_projects = data[data[invoice.PI_FIELD] == pi].copy().reset_index(drop=True)
+
+        # Remove prepay group data if it's empty
+        if pandas.isna(pi_projects[invoice.GROUP_NAME_FIELD]).all():
+            pi_projects = pi_projects.drop(
+                [
+                    invoice.GROUP_NAME_FIELD,
+                    invoice.GROUP_INSTITUTION_FIELD,
+                    invoice.GROUP_BALANCE_FIELD,
+                    invoice.GROUP_BALANCE_USED_FIELD,
+                ],
+                axis=1,
+            )
+
+        # Add a row containing sums for certain columns
+        column_sums = list()
+        sum_columns_list = list()
```

Comment:
In general, prefer `[]` over `list()` to initialize list variables:

```suggestion
        column_sums = []
        sum_columns_list = []
```